### PR TITLE
VR > Sold out message for views.

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://moderntribe.atlassian.net/browse/ET-764
🎥 https://www.loom.com/share/186af3ca76104d1a8943e95ebed92d95

### CODE REVIEW

File: src/views/v2/day/event/cost.php [hunk 1]
```diff
@@ -24,18 +24,23 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/day/event/cost.php:37]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 3615c042fb772e911d3b3f0a3ac061d3</sub>

File: src/views/v2/list/event/cost.php [hunk 2]
```diff
@@ -21,21 +21,25 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/list/event/cost.php:36]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 9aa63f3c811282861e1c74b69235108d</sub>

File: src/views/v2/month/calendar-body/day/calendar-events/calendar-event/tooltip/cost.php [hunk 3]
```diff
@@ -22,14 +22,19 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day/calendar-events/calendar-event/tooltip/cost.php:35]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: c6715a0f427be4ca5b4f4df403e5d4b8</sub>

File: src/views/v2/month/mobile-events/mobile-day/mobile-event/cost.php [hunk 4]
```diff
@@ -27,14 +27,19 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/month/mobile-events/mobile-day/mobile-event/cost.php:40]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 544b2fda3bf1458b94ddf8daff137194</sub>

File: src/views/v2/photo/event/cost.php [hunk 5]
```diff
@@ -22,18 +22,23 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/photo/event/cost.php:35]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: cad12fa66bff00640a7af9c0ed702cb4</sub>

File: src/views/v2/week/grid-body/events-day/event/tooltip/cost.php [hunk 6]
```diff
@@ -22,14 +22,19 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/week/grid-body/events-day/event/tooltip/cost.php:35]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 2a92065ab5d34b787f18517f97f2d2ca</sub>

File: src/views/v2/week/mobile-events/day/event/cost.php [hunk 7]
```diff
@@ -23,18 +23,23 @@
...
 		<a
 			href="<?php echo esc_url( $event->tickets->link->anchor ); ?>"
 			class="tribe-events-c-small-cta__link tribe-common-cta tribe-common-cta--thin-alt"
 		>
 			<?php echo esc_html( $event->tickets->link->label ); ?>
 		</a>
 	<?php endif; ?>
+	<?php if ( $event->tickets->sold_out() ) : ?>
+		<span class="tribe-events-c-small-cta__sold-out tribe-common-b3--bold">
+			<?php echo esc_html( $event->tickets->stock->sold_out ) ?>
```
Comment (tr1b0t) [inline src/views/v2/week/mobile-events/day/event/cost.php:36]: :no_entry_sign: Inline PHP statement must end with a semicolon

<sub>hash: 1333c7c45fc22c0a0ad5e3dc6f232e28</sub>

**Comment by juanfra** [comment]
Thanks!

File: src/Tribe/Events/Views/V2/Models/Tickets.php [hunk 8]
```diff
@@ -145,11 +146,12 @@ public function fetch_data() {
 			}
 
 			if ( ! $data['available'] ) {
-				$parts[ $type . '_stock' ] = esc_html_x( 'Sold out', 'list view stock sold out', 'event-tickets' );
+				$parts[ $type . '_stock' ] = esc_html_x( 'Sold Out', 'new views stock sold out', 'event-tickets' );
```
Comment (sc0ttkclark) [inline src/Tribe/Events/Views/V2/Models/Tickets.php:149]:
```suggestion
				$parts[ $type . '_stock' ] = esc_html_x( 'Sold Out', 'events stock sold out (v2)', 'event-tickets' );
```

File: src/Tribe/Events/Views/V2/Models/Tickets.php [hunk 9]
```diff
@@ -303,4 +306,32 @@ public function exist() {
...
+	public function in_date_range() {
+		if ( ! $this->post_id ) {
+			return false;
+		}
+
+		return tribe_tickets_is_current_time_in_date_window( $this->post_id );
+	}
+
+	/**
+	 * Returns whether an event has its tickets sold out
```
Comment (sc0ttkclark) [inline src/Tribe/Events/Views/V2/Models/Tickets.php:326]:
```suggestion
	 * Returns whether an event has its tickets sold out.
```

**Pull request merged**